### PR TITLE
Add SAPRFC source

### PR DESCRIPTION
**Pull request by AnnaGerlich**
## Summary

- Adds SAPRFC source.
- Adds SAPRFC integration test.
- Updates sap_rfc base image to viadot:2.0a11.

## Importance

Enables SAP RFC connection

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `Databricks` source to the library.
 - Added `ExchangeRates` source to the library.
 - Added `from_df()` method to `Azure Data Lake` source
+- Added `sap_rfc` source to the library.
```
Comment (Trymzet) [inline CHANGELOG.md:13]: `SAPRFC`*

File: viadot/examples/sap_rfc/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM viadot:dev
+FROM viadot:2.0a11
```
Comment (Trymzet) [inline viadot/examples/sap_rfc/Dockerfile:1]: This should be 2.0a12 in fact as a11 version doesn't have `SAPRFC` yet (once this is merged I'll publish this new release).

File: viadot/sources/sap_rfc.py [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+import pandas as pd
+
+try:
+    import pyrfc
+    from pyrfc._exception import ABAPApplicationError
+except ModuleNotFoundError:
+    raise ImportError("pyfrc is required to use the SAPRFC source.")
+from sql_metadata import Parser
+
+from viadot.config import CONFIG
```
Comment (Trymzet) [inline viadot/sources/sap_rfc.py:17]: This has changed in 2.0 (credential handling; see existing sources).

**Review by Trymzet — CHANGES_REQUESTED** [review]
Minor changes required.

File: viadot/examples/sap_rfc/Dockerfile (hunk 2 above)
Comment (AnnaGerlich) [reply viadot/examples/sap_rfc/Dockerfile:1]:
```suggestion
FROM viadot:2.0a12
```

File: viadot/examples/sap_rfc/README.md [hunk 4]
```diff
@@ -1,11 +1,49 @@
...
-To run tests, run eg. `docker exec -it viadot_saprfc_lab pytest tests/integration/test_sap_rfc.py`.
\ No newline at end of file
+Spin it up with the provided `docker-compose`
+```
+docker-compose up -d
+```
+
+You can now open up Jupyter Lab at `localhost:5678`. 
+
+## Config File
```
[the remark this answers is not in the document]
Comment (AnnaGerlich) [reply viadot/examples/sap_rfc/README.md:23]: Updated in the [Updated credentials to config.yaml](https://github.com/dyvenia/viadot/pull/582/commits/7b2dbffe3d8f2aaffee5b9cf72f7c8487413ab61)

File: viadot/sources/sap_rfc.py (hunk 3 above)
Comment (AnnaGerlich) [reply viadot/sources/sap_rfc.py]: Updated in the [Updated credentials to config.yaml](https://github.com/dyvenia/viadot/pull/582/commits/7b2dbffe3d8f2aaffee5b9cf72f7c8487413ab61)

**Pull request merged**